File: Trabajo_10/CPP/Ejercicios/Ejercicio_1.cpp
#include <iostream>
using namespace std;
#include <vector>

template <typename T>
void ordenamiento_vector(vector<T> &vct) {
    int tamaño = vct.size();
    for (int i = 0; i < tamaño - 1; i++) {
        for (int j = 0; j < tamaño - i - 1; j++) {
            if (vct[j] > vct[j + 1]) {
                T temp = vct[j];
                vct[j] = vct[j + 1];
                vct[j + 1] = temp;
            }
        }
    }
}

int main() {
    vector<int> numeritos = {1, 2, 3, 0, 9, 7, 6};
    ordenamiento_vector(numeritos);

    for (int i = 0; i < numeritos.size(); i++){
        cout << numeritos[i];
    }

    cout << "\n" << endl;

    vector<float> numeritos_flotantes = {6.5, 45.7, 85.5, 2.2, 1.1, 2.5, 4.9};
    ordenamiento_vector(numeritos_flotantes);

    for (int i = 0; i < numeritos_flotantes.size(); i++){
        cout << numeritos_flotantes[i];
    }


    return 0;
};

File: Trabajo_11/JAVA/Ejercicio/Ejercicio_1.java
import java.util.ArrayList;
import java.util.List;

public class Main {
    public static <T extends Comparable<T>> void ordenamientoVector(List<T> vct) {
        int tamaño = vct.size();
        for (int i = 0; i < tamaño - 1; i++) {
            for (int j = 0; j < tamaño - i - 1; j++) {
                if (vct.get(j).compareTo(vct.get(j + 1)) > 0) {
                    T temp = vct.get(j);
                    vct.set(j, vct.get(j + 1));
                    vct.set(j + 1, temp);
                }
            }
        }
    }

    public static void main(String[] args) {
        List<Integer> numeritos = new ArrayList<>();
        numeritos.add(1);
        numeritos.add(2);
        numeritos.add(3);
        numeritos.add(0);
        numeritos.add(9);
        numeritos.add(7);
        numeritos.add(6);
        ordenamientoVector(numeritos);

        for (int i = 0; i < numeritos.size(); i++){
            System.out.print(numeritos.get(i));
        }

        System.out.println("\n");

        List<Float> numeritosFlotantes = new ArrayList<>();
        numeritosFlotantes.add(6.5f);
        numeritosFlotantes.add(45.7f);
        numeritosFlotantes.add(85.5f);
        numeritosFlotantes.add(2.2f);
        numeritosFlotantes.add(1.1f);
        numeritosFlotantes.add(2.5f);
        numeritosFlotantes.add(4.9f);
        ordenamientoVector(numeritosFlotantes);

        for (int i = 0; i < numeritosFlotantes.size(); i++){
            System.out.print(numeritosFlotantes.get(i));
        }
    }
}